src/Core/Mesh/TriangleMesh.inl

```diff
@@ -36,55 +32,132 @@ inline TriangleMesh& TriangleMesh::operator=( TriangleMesh&& other ) {
     m_vertexAttribs = std::move( other.m_vertexAttribs );
     m_verticesHandle = std::move( other.m_verticesHandle );
     m_normalsHandle = std::move( other.m_normalsHandle );
+    return *this;
 }
 
-inline void TriangleMesh::clear() {
-    m_vertexAttribs.clear();
-    vertices().clear();
-    normals().clear();
-    m_triangles.clear();
-
-    initDefaultAttribs();
-}
+inline bool TriangleMesh::append( const TriangleMesh& other ) {
+    // check same attributes through names
+    bool sameAttrib = true;
+    const auto& m_attr = m_vertexAttribs.m_attribsIndex;
+    const auto& o_attr = other.m_vertexAttribs.m_attribsIndex;
+    // one way
+    for ( const auto& attr : m_attr )
+    {
+        if ( o_attr.find( attr.first ) == o_attr.end() )
+        {
+            sameAttrib = false;
+            break;
+        }
+    }
+    if ( !sameAttrib )
+        return false;
+    // the other way
+    for ( const auto& attr : o_attr )
+    {
+        if ( m_attr.find( attr.first ) == m_attr.end() )
+        {
+            sameAttrib = false;
+            break;
+        }
+    }
+    if ( !sameAttrib )
+        return false;
 
-inline void TriangleMesh::append( const TriangleMesh& other ) {
+    // now we can proceed, topology first
     const std::size_t verticesBefore = vertices().size();
     const std::size_t trianglesBefore = m_triangles.size();
-
-    vertices().insert( vertices().end(), other.vertices().cbegin(), other.vertices().cend() );
-    ///\todo what about other attribs ?"
-    normals().insert( normals().end(), other.normals().cbegin(), other.normals().cend() );
+    const std::size_t facesBefore = m_faces.size();
     m_triangles.insert( m_triangles.end(), other.m_triangles.cbegin(), other.m_triangles.cend() );
-
-    // Offset the vertex indices in the faces
+    m_faces.insert( m_faces.end(), other.m_faces.cbegin(), other.m_faces.cend() );
+    // Offset the vertex indices in the triangles and faces
     for ( uint t = trianglesBefore; t < m_triangles.size(); ++t )
     {
         for ( uint i = 0; i < 3; ++i )
         {
             m_triangles[t][i] += verticesBefore;
         }
     }
+    for ( uint f = facesBefore; f < m_faces.size(); ++f )
+    {
+        for ( uint i = 0; i < m_faces[f].size(); ++i )
+        {
+            m_faces[f][i] += verticesBefore;
+        }
+    }
+
+    // Deal with all attributes the same way (vertices and normals too)
+    for ( auto& attr : other.m_vertexAttribs.attribs() )

```

Comment:
I would recommend to use a local template helper here, to reduce code duplication. So the loop would look like:
```c++
if ( attr->isFloat() ) { append_attrib<float>(attr); }
if ( attr->isVec2() )  { append_attrib<Vector2>(attr); }
if ( attr->isVec3() )  { append_attrib<Vector3>(attr); }
if ( attr->isVec4() )  { append_attrib<Vector4>(attr); }
```

The helper would be private, and look like
```c++
template <typename T>
void append_attrib (Attrib<T> attr) {
  auto h = m_vertexAttribs.findAttrib<T>( attr->getName() );
  auto& v0 = static_cast<Attrib<T>*>( &( m_vertexAttribs.getAttrib( h ) ) )->data();
  const auto& v1 = static_cast<Attrib<T>*>( attr )->data();
  v0.insert( v0.end(), v1.cbegin(), v1.cend() );
}
```

Reply:
As this is purely internal and meant to avoid code redundancy, I would put the cast in the helper, as the call is a bit shorter.
But both options looks ok to me.